### PR TITLE
converted home page in hooks

### PR DESCRIPTION
**Pull request by Andreson26**
converted the navbar , markatdata, bitcoin charts and the table coins in hooks!

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,72 +1,113 @@
...
+  const handleCurrencyChange = (currency) => {
+    const newCurrencies = currencies.map((newCurrency) => {
       return {
         ...newCurrency,
         isSelected: newCurrency.name === currency.name,
       };
     });
-    this.setState({ currencies: newCurrencies, selectedCurrency: currency });
+    setSelectedCurrency(currency);
+    setCurrencies(newCurrencies);
```
Comment (yoloOnTheBattlefield) [inline src/App.js:77]: can you get away with using just the currencies array?

File: src/components/coinCharts/BarChart/BarChart.js [hunk 2]
```diff
@@ -18,6 +18,15 @@ export default function LineChart({
...
   };
 
+  /*let time =
+        date.getHours() > 12
+          ? `${date.getHours() - 12}:${date.getMinutes()} PM`
+          : `${date.getHours()}:${String(date.getMinutes()).padStart(
+              2,
+              "0"
+            )} AM`;  === 1
+        ? time*/
```
Comment (yoloOnTheBattlefield) [inline src/components/coinCharts/BarChart/BarChart.js:28]: do you need this?

File: src/components/coinTable/CoinsTable/CoinsTable.js [hunk 3]
```diff
@@ -18,55 +18,55 @@ const headerColumns = [
...
+      setIsLoading(true);
       const { data } = await axios(
-        `${apiCoinsURL}markets?vs_currency=${this.props.selectedCurrency.code}&order=market_cap_desc&per_page=10&page=${this.state.page}&sparkline=true&price_change_percentage=1h%2C24h%2C7d`
+        `${apiCoinsURL}markets?vs_currency=${selectedCurrency.code}&order=market_cap_desc&per_page=10&page=${page}&sparkline=true&price_change_percentage=1h%2C24h%2C7d`
       );
-      this.setState({
-        coinList: [...this.state.coinList, ...data],
-        isLoading: false
-      });
+      console.log(data);
```
Comment (yoloOnTheBattlefield) [inline src/components/coinTable/CoinsTable/CoinsTable.js:34]: no console

File: src/components/coinTable/CoinsTable/CoinsTable.js [hunk 4]
```diff
@@ -18,55 +18,55 @@ const headerColumns = [
...
-    this.getAllCoins(this.state.page);
-  }
-
-  componentDidUpdate(prevProps, prevState) {
-    if (prevState.page !== this.state.page) {
-      this.getAllCoins();
-    }
-  }
+  useEffect(
+    (prevState) => {
```
Comment (yoloOnTheBattlefield) [inline src/components/coinTable/CoinsTable/CoinsTable.js:44]: is this a thing?

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 1 above)
Comment (Andreson26) [reply src/App.js:77]: ok

**Pull request merged**